### PR TITLE
LP-1599 Integrate badge progress with frontend

### PR DESCRIPTION
**Pull request by zeshanarifios**
[LP-1599](https://philanthropyu.atlassian.net/browse/LP-1599)

### CODE REVIEW

File: openedx/features/badging/tests/test_model_badge.py [hunk 1]
```diff
@@ -39,77 +29,10 @@ def test_save_badge_normal(self):
                       date_created=None)
         self.assertEqual(badge.save(), None)
 
-    def test_unearned_badges_empty(self):
+    def test_get_badges_json(self):
         """
-        Get unearned badges dictionary without saving any
-        badges earlier so nothing matches
+        Test get badges json method
```
Comment (zahid-arbisoft) [inline openedx/features/badging/tests/test_model_badge.py]: Lets update doc string stating the scenario and how data is generated

Comment (zeshanarifios) [reply openedx/features/badging/tests/test_model_badge.py]: done

File: openedx/features/badging/tests/test_model_badge.py [hunk 2]
```diff
@@ -39,29 +29,10 @@ def test_save_badge_normal(self):
...
         """
-        expected_result = {}
-        returned_result = Badge.get_badges(community_type="no_match_")
+        expected_result = '[]'
+        returned_result = Badge.get_badges_json(badge_type="no_match_")
         self.assertEqual(expected_result, returned_result)
-
-    def test_all_badges(self):
-        """
```
Comment (mabdullahz) [inline openedx/features/badging/tests/test_model_badge.py]: Let's please add a test for badge ordering.

Comment (zeshanarifios) [reply openedx/features/badging/tests/test_model_badge.py]: done

**Pull request merged**